components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSubscriberDAOImpl.java

```diff
@@ -0,0 +1,487 @@
+                    createSubscriberPropertiesPrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+                    createSubscriberPropertiesPrepStmt.setInt(TENANT_ID, tenantId);
+
+                    createSubscriberPropertiesPrepStmt.addBatch();
+                }
+            }
+            createSubscriberPropertiesPrepStmt.executeBatch();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            IdentityDatabaseUtil.rollbackTransaction(connection);
+            throw new EntitlementException("Error while inserting subscriber details", e);
+        } finally {
+            IdentityDatabaseUtil.closeConnection(connection);
+        }
+    }
+
+    /**
+     * DAO method to update a subscriber.
+     *
+     * @param subscriberId        subscriber ID.
+     * @param updatedModuleName   updated module name.
+     * @param updatedPropertyDTOS updated property DTOs.
+     * @param tenantId            tenant ID.
+     * @throws EntitlementException If an error occurs.
+     */
+    private void updateSubscriber(String subscriberId, String updatedModuleName,
+                                  PublisherPropertyDTO[] updatedPropertyDTOS, int tenantId)
+            throws EntitlementException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+        try (NamedPreparedStatement updateSubscriberPropertiesPrepStmt = new NamedPreparedStatement(connection,
+                UPDATE_SUBSCRIBER_PROPERTIES_SQL)) {
+
+            // Update the module name of an existing subscriber
+            if (StringUtils.isNotBlank(updatedModuleName)) {
+                try (NamedPreparedStatement updateSubscriberPrepStmt = new NamedPreparedStatement(connection,
+                        UPDATE_SUBSCRIBER_MODULE_SQL)) {
+                    updateSubscriberPrepStmt.setString(ENTITLEMENT_MODULE_NAME, updatedModuleName);
+                    updateSubscriberPrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+                    updateSubscriberPrepStmt.setInt(TENANT_ID, tenantId);
+                    updateSubscriberPrepStmt.executeUpdate();
+                }
+            }
+
+            // Update the property values of an existing subscriber
+            if (ArrayUtils.isNotEmpty(updatedPropertyDTOS)) {
+                for (PublisherPropertyDTO dto : updatedPropertyDTOS) {
+                    updateSubscriberPropertiesPrepStmt.setString(PROPERTY_VALUE, dto.getValue());
+                    updateSubscriberPropertiesPrepStmt.setString(PROPERTY_ID, dto.getId());
+                    updateSubscriberPropertiesPrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+                    updateSubscriberPropertiesPrepStmt.setInt(TENANT_ID, tenantId);
+                    updateSubscriberPropertiesPrepStmt.addBatch();
+                }
+            }
+            updateSubscriberPropertiesPrepStmt.executeBatch();
```

Comment:
Also this one can be moved inside of the previous if block
`if (ArrayUtils.isNotEmpty(updatedPropertyDTOS)) {`